### PR TITLE
#160127329 Refactor Dockerfile and Makefile

### PR DESCRIPTION
**Pull request by HarithJ**
#### What does this PR do?
- Removes line that copies entire project to docker image used for development
- Builds docker image before running tests

#### Description of Task to be completed?
1. Dockerfile, used for development purposes, is copying entire root folder into the container while it only needs to mount it as a volume. Remove the line that copies entire root folder into the container.

2. Currently, when running tests inside the container using `make test` command, it is NOT building the image, and as a result, if there are any changes to `package.json` file, they would go unnoticed and tests may fail. Update the makefile so that when `make test` command is ran, it also builds the docker image.

#### How should this be manually tested?
Run `make test` command on the terminal and it should first build the docker image.

#### What are the relevant pivotal tracker stories?
#160127329

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -55,6 +55,8 @@ stop:
 
 ## Run project test cases
 test:background
+	@ ${INFO} "Building required docker images"
```
Comment (nyambati) [inline Makefile]: Move this to the background target since its the one that starts the container.

Comment (HarithJ) [reply Makefile]: It is not starting a container, rather it is building an image.

Comment (nyambati) [reply Makefile]: You need to rebuild the image before you start the container right? at this stage the container you are running test in has already been started which means that the image you are building would not be the one you are running tests in.

Comment (HarithJ) [reply Makefile]: I have made the changes.

**Pull request merged**